js/privatebin.js

```diff
@@ -374,6 +374,31 @@ jQuery.PrivateBin = (function($, RawDeflate) {
             }
         };
 
+        /**
+         * formats the text that needs to be formatted, so DomPurify can properly escape it.
+         *
+         * @name   Helper.preformatTextForDomPurify
+         * @function
+         * @param  {string} html
+         * @param  {'markdown'|'syntaxhighlighting'|'plaintext'} text
+         * @return {string} new text
+         */
+        me.preformatTextForDomPurify = function(text, format)
+        {
+            if (!format) {
+                throw new TypeError('invalid format parameter');
+            }
+
+            // encode < to make sure DomPurify does not interpret e.g. HTML or XML markup as code
+            // cf. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/xmp#Summary
+            // As Markdown, by definition, is/allows HTML code, we do not do anything there.
+            if (format !== 'markdown') {
+                // one character is enough, as this is not security-relevant (all output will go through DOMPurify later)
+                text = text.replace(/</g, '&lt;');
```

Comment:
This doesn't encode all of the HTML entities - they should all get displayed unchanged, not just the lighter-then-signs. Couldn't you just use the the existing Helper.htmlEntities() instead of creating a new function?

Reply:
Please see https://zerobin-sqlite.dssr.ch/?74001c1c3e4d9a90#3khg4MG1i7jNWbQSUpXFsnoBRbbGkxuUmVCjqZmpUn1h which uses this branches code - note how all entities except `<` get displayed incorrectly.